drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -571,6 +585,15 @@ local function initialize_switch(driver, device)
     device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
   end
 
+  -- If there is a custom static profile for the device, configure buttons if needed and
+  -- then return to prevent profile from being updated.
+  if check_fingerprint_profile_overrides(device) then
+    if #button_eps > 0 then
+      configure_buttons(device)
+    end
+    return
+  end
```

Comment:
this check is to stop the device from configuring to light-level-button? It seems like this should be handled more dynamically, since I can see many more devices being added that use a SmartPlug category rather than a Light, no?

Reply:
and more than that, devices that don't use switch level. sorry, missed that other difference on my first pass

Reply:
I was debating adding this device to `is_supported_combination_button_switch_device_type`, and then adding a bit of logic to use the `plug-button` profile rather than `light-level-button`, but since there is going to be a specific fingerprint added for this device, I decided to add it to `fingerprint_profile_overrides` and just skip the profile selection logic instead. `fingerprint_profile_overrides` was planning to be added by a different PR anyway so I thought it might be easier this way, but I would be ok with either option if anyone has an opinion on this!

Adding it to `is_supported_combination_button_switch_device_type` would have the benefit of supporting devices with a plug and any number of buttons as well.

Reply:
yeah, I believe the general take on these kinds of questions has been to create the correct profile dynamically (even if there's a fingerprint) since this will future proof this work, will allow non-wwst devices to profile correctly, and will add some more error-prone logic for wwst devices.